postgresql: 13.18 -> 13.20, 14.15 -> 14.17, 15.10 -> 15.12, 16.6 -> 16.8, 17.2 -> 17.4

Ma27 opened the pull request:
Closes #382032

https://www.postgresql.org/about/news/postgresql-173-167-1511-1416-and-1319-released-3015/

This only updates the server packages, hence this can go straight into master
Please note that the client update (#382032) is delayed due to a regression in the release (https://www.postgresql.org/about/news/out-of-cycle-release-scheduled-for-february-20-2025-3016/).

Backport to 24.11 must wait until then since we don't have a libpq package there.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm committed:
libpq: 17.2 -> 17.4

ChangeLog:
https://www.postgresql.org/about/news/postgresql-173-167-1511-1416-and-1319-released-3015/
https://www.postgresql.org/about/news/out-of-cycle-release-scheduled-for-february-20-2025-3016/

Co-authored-by: Maximilian Bosch <maximilian@mbosch.me>
Co-authored-by: Wolfgang Walther <walther@technowledgy.de>

Ma27 committed:
postgresql_13: 13.18 -> 13.20

ChangeLog:
https://www.postgresql.org/about/news/postgresql-173-167-1511-1416-and-1319-released-3015/
https://www.postgresql.org/about/news/out-of-cycle-release-scheduled-for-february-20-2025-3016/

Co-authored-by: Wolfgang Walther <walther@technowledgy.de>

Ma27 committed:
postgresql_14: 14.15 -> 14.17

ChangeLog:
https://www.postgresql.org/about/news/postgresql-173-167-1511-1416-and-1319-released-3015/
https://www.postgresql.org/about/news/out-of-cycle-release-scheduled-for-february-20-2025-3016/

Co-authored-by: Wolfgang Walther <walther@technowledgy.de>

Ma27 committed:
postgresql_15: 15.10 -> 15.12

ChangeLog:
https://www.postgresql.org/about/news/postgresql-173-167-1511-1416-and-1319-released-3015/
https://www.postgresql.org/about/news/out-of-cycle-release-scheduled-for-february-20-2025-3016/

Co-authored-by: Wolfgang Walther <walther@technowledgy.de>

Ma27 committed:
postgresql_16: 16.6 -> 16.8

ChangeLog:
https://www.postgresql.org/about/news/postgresql-173-167-1511-1416-and-1319-released-3015/
https://www.postgresql.org/about/news/out-of-cycle-release-scheduled-for-february-20-2025-3016/

Co-authored-by: Wolfgang Walther <walther@technowledgy.de>

Ma27 committed:
postgresql_17: 17.2 -> 17.4

ChangeLog:
https://www.postgresql.org/about/news/postgresql-173-167-1511-1416-and-1319-released-3015/
https://www.postgresql.org/about/news/out-of-cycle-release-scheduled-for-february-20-2025-3016/

Co-authored-by: Wolfgang Walther <walther@technowledgy.de>

Ma27 commented:
Added a commit for the `postgresqlTestHook` issue, then we can try it out right here.

I decided to not fix up the VM tests (they will also build for `postgresqlForTests` right now) since this will resolve itself when the next staging cycle lands on master.

Ma27 commented:
This push should bring it down a bit.
I left out psycopg2 sicne it seems to pull in postgresql entirely and I'll need to take a closer look. But got some other things to do first.

Ma27 commented:
> ~ 700 are from postgresql + extensions directly. That's the minimum any update to all major versions will always have.

Yeah, right. A little more than I would've guessed :( 

I removed the `postgresqlForTests` patch for now since it doesn't make a real difference.

@vcunat @mweinelt are you OK with staging-next? (~1.8k rebuilds for x86_64-linux - but will be more if we also do libpq here).

Re psql: I guess I'll apply the patch and pull the libpq update here, but this will have to wait until tomorrow.

mweinelt commented:
Yes, send it.

mweinelt commented:
https://www.postgresql.org/message-id/272abbd9-d24c-49f1-8b61-83721906aa3b@postgresql.org

Looks like the security fix introduced a regression.

Ma27 commented:
@wolfgangwalther @mweinelt I prepared the relevant changes yesterday (i.e. cherry-picking #382032 and applying the relevant fix), hence I just pushed it.

I'd be fine with both ways, i.e. waiting until the tag is there or going with it now.

vcunat commented:
> it should be "stamped" (tagged in git) later today already

Still no tags over 24h later.

wolfgangwalther committed:
postgresql: fetch from GitHub instead of tarball

This allows us to fetch newer releases a few days before they are
officially released and the tarballs have been pushed.

The regular release cycle is like this:
- Sunday, the release notes are committed.
- Monday, the version is "stamped", i.e. the version bump committed.
- Thursday, the release is made.

There are not going to be any changes from Monday on, so we can kick off
our builds at that time already - they still need time to hit unstable
anyway.

vcunat commented:
Timeline?  I don't think it matters much.  All linux binaries are ready now.  So even if `staging-next` got merged to `master` before this PR, merging to `master` within days after that would be about the same in terms of rebuilds.

Ma27 commented:
> I hope you don't mind pushing to your PR, @Ma27

All good, would've done the same @wolfgangwalther  :) 

> Can we do this in one go to release-24.11 or do we need to go through staging there as well?

staging given the amount of rebuilds.
I'll hopefully get to it this afternoon.

vcunat commented:
aarch64-darwin is in trouble:
https://hydra.nixos.org/eval/1812132?filter=postgresql\_&compare=1812115

vcunat commented:
The build passes for me.  Maybe it really is flaky.

Ma27 commented:
> @Ma27 are you going to do the backport?

Will do so tomorrow.
